src/qibocal/protocols/characterization/utils.py

```diff
@@ -328,51 +331,27 @@ def fit_punchout(data: Data, fit_type: str):
 
     for qubit in qubits:
         qubit_data = data[qubit]
-        freqs = np.unique(qubit_data.freq)
-        nvalues = len(np.unique(qubit_data[fit_type]))
-        nfreq = len(freqs)
-        signals = np.reshape(qubit_data.signal, (nvalues, nfreq))
+        freqs = qubit_data.freq
+        amps = getattr(qubit_data, fit_type)
+        signal = qubit_data["signal"]
         if data.resonator_type == "3D":
-            peak_freqs = freqs[np.argmax(signals, axis=1)]
+            mask_freq, mask_amps = extract_feature(
+                freqs, amps, signal, "max", ci_first_mask=90
```

Comment:
```suggestion
                freqs, amps, signal, "max",
```
Should work also like this, right?

Reply:
Yes, but the  default value of 'ci_first_mask' is 99, in this case the fit was not working properly, because a small amount of points were taken.

Reply:
Oh yes I didn't notice it.